### PR TITLE
Fix controller starting with later load of robot description test

### PR DESCRIPTION
**Pull request by saikishor**
Fixes the failing tests in the ros2_control_ci jobs: https://github.com/ros-controls/ros2_control_ci/issues/98 after the recent merge of https://github.com/ros-controls/ros2_control/pull/1358. The following tests is failing `TestLoadControllerWithoutRobotDescription.controller_starting_with_later_load_of_robot_description`. This PR tries to address this as well as some minor adjustments needed to clean the code

Fixes: https://github.com/ros-controls/ros2_control_ci/issues/98
Fixes: #1620

### CODE REVIEW

File: controller_manager/test/test_spawner_unspawner.cpp [hunk 1]
```diff
@@ -327,6 +327,61 @@ TEST_F(TestLoadController, unload_on_kill)
   ASSERT_EQ(cm_->get_loaded_controllers().size(), 0ul);
 }
 
+TEST_F(TestLoadController, spawner_test_fallback_controllers)
```
Comment (bmagyar) [inline controller_manager/test/test_spawner_unspawner.cpp:330]: Why was this moved?

File: controller_manager/src/controller_manager.cpp [hunk 2]
```diff
@@ -220,20 +220,7 @@ ControllerManager::ControllerManager(
...
-
-  if (is_resource_manager_initialized())
-  {
-    init_services();
-  }
-
-  diagnostics_updater_.setHardwareID("ros2_control");
-  diagnostics_updater_.add(
-    "Controllers Activity", this, &ControllerManager::controller_activity_diagnostic_callback);
+  init_controller_manager();
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp:223]: Did we have all this duplicated code her etue whole time?

File: controller_manager/src/controller_manager.cpp [hunk 3]
```diff
@@ -305,6 +295,10 @@ void ControllerManager::robot_description_callback(const std_msgs::msg::String &
   init_resource_manager(robot_description_);
   if (is_resource_manager_initialized())
   {
+    RCLCPP_INFO(
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp:298]: Isn't this a debug print?

File: controller_manager/test/test_spawner_unspawner.cpp (hunk 1 above)
Comment (saikishor) [reply controller_manager/test/test_spawner_unspawner.cpp:330]: I moved it for better log tracking as the newly added test is placed in between an original test. 

https://github.com/ros-controls/ros2_control/blob/9d1aff96291302c6db153f863792350e352f66a9/controller_manager/test/test_spawner_unspawner.cpp#L330-L407

File: controller_manager/src/controller_manager.cpp (hunk 2 above)
Comment (saikishor) [reply controller_manager/src/controller_manager.cpp:223]: Well only from the #1358. So it is not so long

**Review by bmagyar — CHANGES_REQUESTED** [review]
A few small questions, sorry for the vagueness, can't do proper checks from the phone

File: controller_manager/src/controller_manager.cpp (hunk 3 above)
Comment (saikishor) [reply controller_manager/src/controller_manager.cpp:298]: yes! it is. I think it is very helpful to know somehow via log that it initialized properly and we are going to start the services. If you want, I can remove it as well. I'm fine with anything

**Comment by saikishor** [comment]
> A few small questions, sorry for the vagueness, can't do proper checks from the phone

Done. Thanks for reviewing it quickly

**Review by christophfroehlich — APPROVED** [review]
Thanks for dealing with it. May also fix #1620

**Comment by saikishor** [comment]
> Thanks for dealing with it. May also fix #1620 

Yes! It does fix #1620

**Review by bmagyar — APPROVED** [review]
Thanks for the great work!
